caddyhttp/basicauth/setup_test.go

```diff
@@ -70,6 +70,30 @@ md5:$apr1$l42y8rex$pOA2VJ0x/0TwaFeAF9nX61`
 		}`, false, "pwd", []Rule{
 			{Username: "user", Resources: []string{"/resource1", "/resource2"}},
 		}},
+		{`basicauth user pwd {
+			path /resource1
+			path /resource2
+		}`, false, "pwd", []Rule{
+			{Username: "user", Resources: []string{"/resource1", "/resource2"}},
+		}},
+		{`basicauth user pwd {
+			/resource1
+			realm "Secure resources"
+			path /resource2
```

Comment:
The path/realm syntax is slightly awkward... not bad, but I wonder if we could improve it. What if we moved the realm up to the first line as an optional third argument?

```plain
basicauth user pwd realm {
    paths...
}
```

That way we wouldn't need the `path`/`realm` subdirectives and you can't specify the realm more than once.